can't get sizeof T, error: ';' expected

### Describe the bug

when use sizeof(T), it will cause v compiler error.

### Reproduction Steps

bb.v

```v
const cache_line_size = 64

struct PaddedSlot[T] {
mut:
    data T
    pad [cache_line_size - sizeof(T)]u8
}

fn main() {

        x := PaddedSlot[int]{}
        dump(x)
}
```

```
v bb.v
```

### Expected Behavior

should compile

### Current Behavior

```
$ v bb.v
================== C compilation error (from tcc): ==============
cc: /tmp/v_1000/bb.01JYV77CMQWYC9YHAW07YZ07A6.tmp.c:1064: error: ';' expected (got "Array_fixed_u8_0")
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
builder error:
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
```

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V 0.4.11 6b45931598fd8e405fd53f8fa35d72be639b73e1.d1d43ab

### Environment details (OS name and version, etc.)

|V full version      |V 0.4.11 6b45931598fd8e405fd53f8fa35d72be639b73e1.d1d43ab
|:-------------------|:-------------------
|OS                  |linux, Ubuntu 24.04.2 LTS
|Processor           |8 cpus, 64bit, little endian, Intel(R) Core(TM) i7-9700 CPU @ 3.00GHz
|Memory              |7.81GB/15.51GB
|                    |
|V executable        |/media/HD/github/kbkpbot/v/v
|V last modified time|2025-06-27 13:18:51
|                    |
|V home dir          |OK, value: /media/HD/github/kbkpbot/v
|VMODULES            |OK, value: /home/mars/.vmodules
|VTMP                |OK, value: /tmp/v_1000
|Current working dir |OK, value: /home/mars/v/bug/lfrb
|                    |
|Git version         |git version 2.43.0
|V git status        |weekly.2025.17-319-ga512709e
|.git/config present |true
|                    |
|cc version          |cc (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0
|gcc version         |gcc (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0
|clang version       |Ubuntu clang version 18.1.3 (1ubuntu1)
|tcc version         |tcc version 0.9.28rc 2025-02-13 HEAD@f8bd136d (x86_64 Linux)
|tcc git status      |thirdparty-linux-amd64 696c1d84
|emcc version        |N/A
|glibc version       |ldd (Ubuntu GLIBC 2.39-0ubuntu8.4) 2.39

> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/checker/comptime.v
+++ b/vlib/v/checker/comptime.v
@@ -402,7 +402,7 @@ fn (mut c Checker) eval_comptime_const_expr(expr ast.Expr, nlevel int) ?ast.Comp
 			}
 		}
 		ast.SizeOf {
-			s, _ := c.table.type_size(expr.typ)
+			s, _ := c.table.type_size(c.unwrap_generic(expr.typ))
 			return i64(s)
 		}
 		ast.FloatLiteral {

--- a/vlib/v/checker/struct.v
+++ b/vlib/v/checker/struct.v
@@ -939,6 +939,16 @@ or use an explicit `unsafe{ a[..] }`, if you do not want a copy of the slice.',
 		} else if struct_sym.info.generic_types.len == struct_sym.info.concrete_types.len {
 			parent_type := struct_sym.info.parent_type
 			parent_sym := c.table.sym(parent_type)
+			if c.inside_generic_struct_init {
+				mut st := unsafe { struct_sym.info }
+				for mut field in st.fields {
+					sym := c.table.sym(field.typ)
+					if sym.info is ast.ArrayFixed && c.array_fixed_has_unresolved_size(sym.info) {
+						mut size_expr := unsafe { sym.info.size_expr }
+						field.typ = c.eval_array_fixed_sizes(mut size_expr, 0, sym.info.elem_type)
+					}
+				}
+			}
 			for method in parent_sym.methods {
 				generic_names := struct_sym.info.generic_types.map(c.table.sym(it).name)
 				for i, param in method.params {

--- a/vlib/v/tests/structs/struct_generic_sizeof_test.v
+++ b/vlib/v/tests/structs/struct_generic_sizeof_test.v
@@ -1,0 +1,20 @@
+const cache_line_size = 32
+
+struct PaddedSlot[T] {
+mut:
+	data T
+	pad  [cache_line_size - sizeof(T)]u8
+}
+
+fn test_main() {
+	x := PaddedSlot[int]{}
+	assert '${x}' == 'PaddedSlot[int]{
+    data: 0
+    pad: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
+}'
+	x2 := PaddedSlot[u8]{}
+	assert '${x2}' == 'PaddedSlot[u8]{
+    data: 0
+    pad: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
+}'
+}

